src/api.js

```
@@ -0,0 +1,107 @@
+import axios from 'axios';
+
+/**
+ * ✅ Base URL exigida pela Mate Academy
+ * Caso o .env contenha VITE_API_BASE_URL, ele será usado.
+ * Caso contrário, usa a Mate API padrão.
+ */
+export const BASE_URL =
+  import.meta.env.VITE_API_BASE_URL || 'https://mate-academy.github.io/fe-students-api';
+
+/**
+ * Configuração do cliente Axios
+ */
+const api = axios.create({
+  baseURL: BASE_URL,
+  timeout: 12000,
+});
+
+/**
+ * ===============================
+ * USERS
+ * ===============================
+ */
+
+/**
+ * Busca todos os usuários
+ */
+export async function fetchUsers() {
+  const { data } = await api.get('/users');
```

All these wrapper functions currently `await` the axios call and return `data` which makes callers responsible for catching errors. This is OK, but decide on a consistent approach: either let errors bubble (and handle them in components) or centralise handling here (try/catch -> throw a normalized error). Document the chosen approach so component code implements loaders/errors correctly per requirements.